### PR TITLE
Detect AVX capabilities of the host under msvc

### PR DESCRIPTION
**Pull request by andrewevstyukhin**

### CODE REVIEW

File: tools/simd.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
+include(CheckCXXSourceRuns)
```
Comment (dendibakh) [inline tools/simd.cmake:1]: I suggest renaming the file to something like `msvc_isa.cmake` since it only affects the MSVC compiler.

**Review by dendibakh — COMMENTED** [review]
There is only a problem for the MSVC compiler that you're trying to solve, correct? And the issue is that by default (without `/arch` option) the MSVC compiler will generate SSE code, right?

If yes, then I'm OK with the change, but please rename the `simd.cmake` file.

**Comment by dendibakh** [comment]
@andrewevstyukhin, ping

**Comment by andrewevstyukhin** [comment]
pong
> c:\sde-external-9.0.0-2021-11-07-win\sde.exe **-snb** -- cmake -DCMAKE_BUILD_TYPE=Release ..
-- Building for: Visual Studio 16 2019
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.22000.
-- The C compiler identification is MSVC 19.29.30139.0
-- The CXX compiler identification is MSVC 19.29.30139.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
**-- Performing Test SUPPORT_MSVC_AVX512
-- Performing Test SUPPORT_MSVC_AVX512 - Failed
-- Performing Test SUPPORT_MSVC_AVX2
-- Performing Test SUPPORT_MSVC_AVX2 - Failed
-- Performing Test SUPPORT_MSVC_AVX
-- Performing Test SUPPORT_MSVC_AVX - Success**
-- Configuring done
-- Generating done

**Comment by andrewevstyukhin** [comment]
> cmake -DCMAKE_BUILD_TYPE=Release ..
-- Building for: Visual Studio 16 2019
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.22000.
-- The C compiler identification is MSVC 19.29.30139.0
-- The CXX compiler identification is MSVC 19.29.30139.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
**-- Performing Test SUPPORT_MSVC_AVX512
-- Performing Test SUPPORT_MSVC_AVX512 - Success
-- Performing Test SUPPORT_MSVC_AVX2
-- Performing Test SUPPORT_MSVC_AVX2 - Success
-- Performing Test SUPPORT_MSVC_AVX
-- Performing Test SUPPORT_MSVC_AVX - Success**
-- Configuring done
-- Generating done

**Review by dendibakh — APPROVED** [review]

**Pull request merged**